### PR TITLE
fixed: #5233 firefox related issue for backspace deletion  where "focus" offset is of type element instead of text

### PR DESCRIPTION
**Pull request by Sahejkm**
https://github.com/facebook/lexical/assets/163521239/944ae9bf-1eed-49af-94fe-bc37305624c7

### CODE REVIEW

**Review by zurfyx — COMMENTED** [review]
Thank you!

First of, the selection is not necessarily incorrect. It is a valid selection. Can we do better? Probably, the result you've proposed seems reasonable to me but I would inclined to do a bit more research, particularly since this is a core function.

My hunch is that the normalization "error" we want to address is that the node next to it is a TextNode and this selection becomes redundant, so by moving to the end of the TextNode this selection is better. This already happens in the opposite case LTR (L108-109) but unfortunately for the other case we cut short.

However, before doing any of this, I would be inclined to dig deeper into the actual root cause. Why is this selection causing an issue with deleteCharacter? Can we look into `deleteCharacter(isBackward: boolean)`?

Finally, I would add an E2E test to cover this, we have E2E tests that automatically run in the Chrome, FF, and Safari so hopefully Firefox on Playwright behaves just like the browser version and we can test against it.

**Comment by zurfyx** [comment]
> Fixes #5233
> 
> @zurfyx what is $normalizeSelection and should it still be experimental?

We can probably make it official, initially the idea was to keep this sort of private but I've found that it is a handy shortcut at times when implementing plugins.

**Comment by Sahejkm** [comment]
> However, before doing any of this, I would be inclined to dig deeper into the actual root cause. Why is this selection causing an issue with deleteCharacter? Can we look into `deleteCharacter(isBackward: boolean)`?

i did look into deleteCharacter method
it calls "this.removeText();" by doing "insertText()"
so other approach is we could keep the change local to that delete function, something like

```
   if(this.anchor.isBefore(this.focus) && this.focus.type === 'element' && $isLineBreakNode(this.focus.getNode().getChildAtIndex(this.focus.offset)) ){
      const offset = this.focus.offset;
      const node = this.focus.getNode(); 
      const nextNode = node.getChildAtIndex(offset - 1);
      if(nextNode != null){
      this.focus.set(
        nextNode.__key,
        nextNode.getTextContentSize(),
        'text',
      )
      }
    }
    this.removeText();
```

This also works as original issue came from line 
> firstNode = firstNode.spliceText(startOffset, delCount, text, true);

the issue is that endOffset and delCount in the method get calculated wrongly given the endOffset is from focusNodeOffset which is of value 3 as its of type "element" instead of type TextNode

File: packages/lexical-playground/__tests__/e2e/Selection.spec.mjs [hunk 1]
```diff
@@ -558,4 +559,44 @@ test.describe('Selection', () => {
       `,
     );
   });
+
+  test('can delete word at end of line in a multi paragraph with linebreak by backspace', async ({
+    page,
+    isPlainText,
+  }) => {
+    test.skip(isPlainText);
```
Comment (zurfyx) [inline packages/lexical-playground/__tests__/e2e/Selection.spec.mjs:567]: It reproes on plain text too `?https://playground.lexical.dev/?isRichText=false`, should we delete this? 
```suggestion
```

File: packages/lexical-playground/__tests__/e2e/Selection.spec.mjs [hunk 2]
```diff
@@ -558,4 +559,44 @@ test.describe('Selection', () => {
...
+    // enter linebreak element
+    await pressShiftEnter(page);
+    await page.keyboard.type('This is testing three');
+    // enter linebreak element
+    await pressShiftEnter(page);
+    await page.keyboard.press('ArrowLeft');
+    await page.keyboard.press('ArrowUp');
+    await page.keyboard.press('ArrowLeft');
+    await page.keyboard.press('ArrowLeft');
+    await page.keyboard.press('ArrowLeft');
```
Comment (zurfyx) [inline packages/lexical-playground/__tests__/e2e/Selection.spec.mjs:582]:
```suggestion
    await moveLeft(page, 3);
```

File: packages/lexical/src/LexicalSelection.ts [hunk 3]
```diff
@@ -822,7 +822,7 @@ export class RangeSelection implements BaseSelection {
       }
     }
 
-    if (selectedNodesLength === 1) {
+    if (anchor.key === focus.key && selectedNodesLength === 1) {
```
Comment (zurfyx) [inline packages/lexical/src/LexicalSelection.ts:825]: Nit, resolving properties at the end of the condition
```suggestion
    if (selectedNodesLength === 1 && anchor.key === focus.key) {
```

**Review by zurfyx — APPROVED** [review]
Added some nits but overall LGTM, thank you Sahej!

**Review by zurfyx — APPROVED** [review]

**Pull request merged**